MAINTAINERS.md

```diff
@@ -0,0 +1,16 @@
+# Maintainers
```

Comment:
do we need this file? https://github.com/orgs/aya-rs/people is the source of truth (though we should make everyone's membership public)